blueprints/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_3_materials.py

```diff
@@ -0,0 +1,132 @@
+"""Module including all formulas from chapter 3 - Materials of NEN-EN 1992-1-1+C2:2011."""
+# pylint: disable=arguments-differ
+import numpy as np
+
+from blueprints.codes.eurocode.nen_en_1992_1_1_c2_2011 import NEN_EN_1992_1_1_C2_2011
+from blueprints.codes.formula import Formula
+from blueprints.type_alias import MPA
+
+
+class Form3Dot1EstimationConcreteCompressiveStrength(Formula):
+    """Class representing formula 3.1 for the estimation of the concrete compressive strength, fcm(t),  after t days
+    with an average temperature of 20 degrees Celsius [MPa]."""
+
+    label = "3.1"
+    source_document = NEN_EN_1992_1_1_C2_2011
+
+    def __init__(
+        self,
+        beta_cc_t: float,
+        f_cm: MPA,
+    ) -> None:
+        """[fcm(t)] The estimated concrete compressive strength [MPa].
+
+        NEN-EN 1992-1-1+C2:2011 art.3.1.2(6) - Formula (3.1)
+
+        Parameters
+        ----------
+        beta_cc_t: float
+            [βcc(t)] Coefficient dependent of the age of concrete [-].
+        f_cm: float
+            [fcm] Average concrete compressive strength on day 28 based on table 3.1 [MPa].
+        """
+        super().__init__()
+        self.beta_cc_t = beta_cc_t
+        self.f_cm = f_cm
+
+    @staticmethod
+    def _evaluate(
+        beta_cc_t: float,
+        f_cm: MPA,
+    ) -> MPA:
+        """Evaluates the formula, for more information see the __init__ method"""
+        if beta_cc_t < 0:
+            raise ValueError(f"Negative beta_cc_t: {beta_cc_t}. beta_cc_t cannot be negative")
```

Comment:
I do think that the basic valuation of the inputs should be in _evaluate. This is to prevent that off values are going in unseen. beta_cc_t is not a formula in this context but an argument.